ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -77,6 +92,217 @@ tests = testGroup "Mempool"
+-- and 'LedgerState'.
+prop_pure_getCapacity :: MempoolCapTestSetup -> Property
+prop_pure_getCapacity (MempoolCapTestSetup TestSetupWithTxs {..}) =
+  withInternalState testSetup $
+  \ _mpArgs internalSt _ledgerState ->
+      let actualCapacity = isCapacity internalSt
+      in  actualCapacity === testCapacity
+  where
+    MempoolCapacityBytesOverride testCapacity = testMempoolCapOverride testSetup
+
+-- | Test the correctness of 'tryAddTxs' when the Mempool is (or will be) at
+-- capacity.
+--
+-- Ignore the "100% empty Mempool" label in the test output, that is there
+-- because we reuse 'withTestMempool' and always start with an empty Mempool
+-- and 'LedgerState'.
+prop_pure_Capacity :: MempoolCapTestSetup -> Property
+prop_pure_Capacity (MempoolCapTestSetup TestSetupWithTxs {..}) =
+  withInternalState testSetup $
+  \ mpArgs internalSt _ledgerState ->
+      let capacity = isCapacity internalSt
+          snapshot = implSnapshotFromIS internalSt
+          curSize  = (msNumBytes . snapshotMempoolSize) snapshot
+          (res@(processed, unprocessed),_internalSt') =
+                     runTryAddTxs mpArgs internalSt (map fst txs)
+      in  counterexample ("Initial size: " <> show curSize)    $
+          classify (null processed)   "no transactions added"  $
+          classify (null unprocessed) "all transactions added" $
+          blindErrors res === expectedResult capacity curSize
+    where
+    -- | Convert 'MempoolAddTxResult' into a 'Bool':
+    -- isMempoolTxAdded -> True, isMempoolTxRejected -> False.
+    blindErrors
+      :: ([(GenTx TestBlock, MempoolAddTxResult blk)], [GenTx TestBlock])
+      -> ([(GenTx TestBlock, Bool)], [GenTx TestBlock])
+    blindErrors (processed, toAdd) = (processed', toAdd)
+      where
+        processed' = [ (tx, isMempoolTxAdded txAddRes)
+                     | (tx, txAddRes) <- processed ]
+
+    expectedResult
+      :: MempoolCapacityBytes
+      -> Word32  -- ^ Current mempool size
+      -> ([(GenTx TestBlock, Bool)], [GenTx TestBlock])
+    expectedResult (MempoolCapacityBytes capacity) = \curSize ->
+        go curSize [] txs
+      where
+        go
+          :: Word32
+          -> [(GenTx TestBlock, Bool)]
+          -> [(GenTx TestBlock, Bool)]
+          -> ([(GenTx TestBlock, Bool)], [GenTx TestBlock])
+        go curSize processed = \case
+          []
+            -> (reverse processed, [])
+          (tx, valid):txsToAdd'
+            | let curSize' = curSize + txSize tx
+            , curSize' <= capacity
+            -> go (if valid then curSize' else curSize)
+                  ((tx, valid):processed)
+                  txsToAdd'
+            | otherwise
+            -> (reverse processed, tx:map fst txsToAdd')
+
+{-------------------------------------------------------------------------------
+  Mempool Pure Test Infrastructure
+-------------------------------------------------------------------------------}
+
+data TestSetupSub = TestSetupSub
```

Comment:
The name `TestSetupSub` is not self-explanatory. You could rename it to something clearer or at least add a comment explaining what it is.